mdp/linear_flows_online.py

```diff
@@ -0,0 +1,607 @@
+from builtins import str
+import mdp
+from mdp import numx
+from .linear_flows import FlowException, FlowExceptionCR, _sys, _traceback
+from collections import deque as _deque
+
+
+class OnlineFlowException(mdp.MDPException):
+    """Base class for exceptions in Flow subclasses."""
+    pass
+
+
+class OnlineFlow(mdp.Flow):
+    """An 'OnlineFlow' is a sequence of online/executable nodes that are trained and executed
+    together to form a more complex algorithm.  Input data is sent to the
+    first node and is successively processed by the subsequent nodes along
+    the sequence.
+
+    Using an online flow as opposed to handling manually a set of nodes has a
+    clear advantage: The general online flow implementation automatizes the
+    training (including supervised training and multiple training phases),
+    execution, and inverse execution (if defined) of the whole sequence.
+
+    OnlineFlow sequence can contain trained or non-trainable Nodes and optionally
+    a terminal trainable Node.
+
+    Differences between a Flow and an OnlineFlow:
+    a) In Flow, data is processed sequentially, training one node at a time. Whereas, in an
+       OnlineFlow data is processed simultaneously training all the nodes at the same time.
+
+    b) Flow requires a list of dataiterables with a length equal to the
+       number of nodes or a single numpy array. OnlineFlow requires only one
+       input dataiterable as each node is trained simultaneously.
+
+    c) Additional train args (supervised labels etc) are passed to each node through the
+       node specific dataiterable. OnlineFlow requires the dataiterable to return a list
+       that contains tuples of args for each node: [x, (node0 args), (node1 args), ...]. See
+       train docstring.
+
+    d) OnlineFlow also has the cache attribute that retrieves any caches variables
+       stored in the OnlineNodes.
```

Comment:
What does this cache exactly do? Is it coupled to the concept of online learning, or is here because it was useful for some specific case?